crates/ruff_server/src/trace.rs

```diff
@@ -48,13 +52,24 @@ pub(crate) fn init_tracing(
         .set(sender)
         .expect("logging sender should only be initialized once");
 
-    let log_file = log_file.and_then(|path| {
-        std::fs::OpenOptions::new()
-            .create(true)
-            .append(true)
-            .open(path)
-            .ok()
-    });
+    let log_file = log_file
+        .map(|path| {
+            if let Some(expanded) = shellexpand::full(&path.to_string_lossy())
+                .ok()
+                .and_then(|path| PathBuf::from_str(&path).ok())
+            {
+                expanded
+            } else {
+                path.to_path_buf()
+            }
+        })
```

Comment:
If not already done, let's document the fact that the path in `logFile` is expanded if it contained `~` (tilde) or any environment variables.

Reply:
Sorry for not being clear. I meant to document it in user-facing docs.